base_layer/core/src/proof_of_work/monero_rx.rs

```diff
@@ -231,25 +231,343 @@ fn verify_root(monero_data: &MoneroData) -> Result<(), MergeMineError> {
     Ok(())
 }
 
-fn merged_mining_subfield(header: &BlockHeader) -> SubField {
-    let hash = header.merged_mining_hash();
-    let depth = 0;
-    SubField::MergeMining(VarInt(depth), Hash::from(from_slice(&hash)))
-}
-
 fn verify_header(header: &BlockHeader, monero_data: &MoneroData) -> Result<(), MergeMineError> {
-    if !(monero_data
-        .coinbase_tx
-        .prefix
-        .extra
-        .0
-        .contains(&merged_mining_subfield(header)))
-    {
+    let mut found = false;
+    for item in monero_data.coinbase_tx.prefix.extra.0.clone() {
+        match item {
+            SubField::MergeMining(x, y) => {
+                let compare = Hash::from(from_slice(&header.merged_mining_hash()));
```

Comment:
What is `compare`? Name it `merge_mining_hash`. This should be moved out of the loop so that it doesn't have to be recalculated every iteration.